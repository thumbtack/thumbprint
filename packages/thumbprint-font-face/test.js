const sass = require('node-sass');
const fs = require('fs');
const importer = require('node-sass-tilde-importer');
const tmp = require('tmp');

it('throws error if variable is not provided correctly', () => {
    let error = null;

    try {
        sass.renderSync({
            file: 'packages/thumbprint-font-face/_index.scss',
            importer,
        }).css.toString();
    } catch (e) {
        error = e;
    }

    expect(error).not.toBeNull();
});

it('works if variable is provided', () => {
    const file = tmp.fileSync({ dir: process.cwd() });

    fs.writeFileSync(
        file.name,
        `$thumbprint-font-url: 'https://example.com/fonts/';\n${fs.readFileSync(
            'packages/thumbprint-font-face/_index.scss',
            'utf-8',
        )}`,
    );

    const css = sass
        .renderSync({
            file: file.name,
            importer,
        })
        .css.toString();
    expect(css).toMatchSnapshot();

    file.removeCallback();
});
