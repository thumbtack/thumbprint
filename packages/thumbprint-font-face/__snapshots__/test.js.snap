// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works if variable is provided 1`] = `
"@font-face {
  font-family: 'Mark';
  font-weight: 400;
  font-display: swap;
  src: local(\\"Mark, Avenir, Helvetica, Arial, sans-serif\\"), url(https://example.com/fonts/mark/mark-tt-subset.woff2) format(\\"woff2\\"), url(https://example.com/fonts/mark/mark-tt-subset.woff) format(\\"woff\\"); }

@font-face {
  font-family: 'Mark';
  font-weight: 700;
  font-display: swap;
  src: local(\\"Mark, Avenir, Helvetica, Arial, sans-serif\\"), url(https://example.com/fonts/mark/mark-tt-subset-bold.woff2) format(\\"woff2\\"), url(https://example.com/fonts/mark/mark-tt-subset-bold.woff) format(\\"woff\\"); }
"
`;
